apis/pipelines/v1alpha5/conversion.go

```diff
@@ -1,55 +1,87 @@
 package v1alpha5
 
 import (
-	"github.com/sky-uk/kfp-operator/apis"
+	hub "github.com/sky-uk/kfp-operator/apis/pipelines/v1alpha6"
 )
 
-type RunConfigurationConversionRemainder struct {
-	RunConversionRemainder `json:",inline"`
-	Triggers               Triggers `json:"triggers,omitempty"`
-}
-
-func (rcr RunConfigurationConversionRemainder) Empty() bool {
-	return len(rcr.Triggers.Schedules) == 0 && len(rcr.Triggers.OnChange) == 0 && rcr.RunConversionRemainder.Empty()
-}
-
-type RunConversionRemainder struct {
-	Artifacts           []OutputArtifact   `json:"artifacts,omitempty"`
-	ValueFromParameters []RuntimeParameter `json:"valueFromParameters,omitempty"`
-}
-
-func (rcr RunConversionRemainder) Empty() bool {
-	return len(rcr.Artifacts) == 0 && len(rcr.ValueFromParameters) == 0
+func convertRuntimeParametersTo(rtp []RuntimeParameter) []hub.RuntimeParameter {
+	convertValueFromTo := func(v *ValueFrom) *hub.ValueFrom {
+		if v != nil {
+			return &hub.ValueFrom{
+				RunConfigurationRef: hub.RunConfigurationRef{
+					Name:           v.RunConfigurationRef.Name,
+					OutputArtifact: v.RunConfigurationRef.OutputArtifact,
+				},
+			}
+		}
+		return nil
+	}
+	var hubRtp []hub.RuntimeParameter
+	for _, namedValue := range rtp {
+		hubRtp = append(hubRtp, hub.RuntimeParameter{
+			Name:      namedValue.Name,
+			Value:     namedValue.Value,
+			ValueFrom: convertValueFromTo(namedValue.ValueFrom),
+		})
+	}
+	return hubRtp
 }
 
-func (rcr RunConversionRemainder) ConversionAnnotation() string {
-	return GroupVersion.Version + "." + GroupVersion.Group + "/conversions.remainder"
+func convertArtifactsTo(oa []OutputArtifact) []hub.OutputArtifact {
+	var hubOa []hub.OutputArtifact
+	for _, artifact := range oa {
+		hubOa = append(hubOa, hub.OutputArtifact{
+			Name: artifact.Name,
+			Path: hub.ArtifactPath{
+				Locator: hub.ArtifactLocator{
+					Component: artifact.Path.Locator.Component,
+					Artifact:  artifact.Path.Locator.Artifact,
+					Index:     artifact.Path.Locator.Index,
+				},
+				Filter: artifact.Path.Filter,
+			},
+		})
+	}
+	return hubOa
 }
 
-func SplitRunTimeParameters(rts []RuntimeParameter) (namedValues []apis.NamedValue, valueFroms []RuntimeParameter) {
-	for _, rt := range rts {
-		if rt.ValueFrom != nil {
-			valueFroms = append(valueFroms, rt)
-		} else {
-			namedValues = append(namedValues, apis.NamedValue{
-				Name:  rt.Name,
-				Value: rt.Value,
-			})
+func convertRuntimeParametersFrom(hubRtp []hub.RuntimeParameter) []RuntimeParameter {
+	convertValueFromFrom := func(v *hub.ValueFrom) *ValueFrom {
```

Comment:
confusing naming or just a mistake? `FromFrom`

Reply:
moved out from an anonymous function and renamed to `convertFromHubValueFrom`

Reply:
We opted out of using the word "hub" everywhere to be follow the naming convention of the interface.